Lib/tkinter/ttk.py

```diff
@@ -1151,6 +1151,28 @@ def __init__(self, master=None, **kw):
         Widget.__init__(self, master, "ttk::sizegrip", kw)
 
 
+class Spinbox(Combobox):
+    """Ttk Spinbox is an Entry with increment and decrement arrows
+
+    It is commonly used for number entry or to select from a list of
+    string values.
+    """
+
+    def __init__(self, master=None, **kw):
+        """Construct a Ttk Spinbox widget with the parent master.
+
+        STANDARD OPTIONS
+
+            class, cursor, style, takefocus, validate,
+            validatecommand, xscrollcommand, invalidcommand
+
+        WIDGET-SPECIFIC OPTIONS
+
+            to, from_, increment, values, wrap, format, command
+        """
+        Entry.__init__(self, master, "ttk::spinbox", **kw)
```

Comment:
Should this be `Combobox.__init__` now?

Reply:
Quite right.  Is there a particular reason why none of the classes in this library use `super()`?

Reply:
I'm actually going to revert the base class to Entry per @terryjreedy 's comment.

Reply:
I am pretty sure super came years later than the original Tkinter module.  In 2.x it was a bit harder to use.  For single inheritance, I don't think that there is much advantage.  (I believe that there are cases of multiple inheritance in tkinter.__init__, but have not rechecked.)

Reply:
I agree this should have a base class of Entry, but with that, I also think `current` and `set` from Combobox should be included here.  It would probably be overkill, but maybe Spinbox and Combobox should both inherit from an Entry subclass that has `current` and `set`?

The tests will be able to use `current` and `set`.

Reply:
The `current` and `set` methods correspond the `current` and `set` subcommands of Ttk widgets. The ttk::entry widget doesn't have these commands.

Reply:
Sorry, I meant that both Spinbox and Combobox have `current` and `set`.  I didn't know if it was better to implement them both directly from Entry or to have another class in between, that is a subclass of Entry that implements current and set and is the superclass to Combobox and Spinbox.